specification/search/data-plane/Azure.Search/preview/2024-03-01-Preview/searchservice.json

```diff
@@ -6184,6 +6209,12 @@
           "type": "string",
           "description": "The name of the kind of vectorization method being configured for use with vector search.",
           "x-nullable": false
+        },
+        "compression": {
+          "x-ms-client-name": "compressionConfigurationName",
+          "type": "string",
+          "description": "The name of the compression method configuration that specifies the compression method and optional parameters.",
```

Comment:
This makes it sound like it's a discriminated field. Is that the case? Are the possible values known? What would customers do with this?

Reply:
I see below there's a `name` property, but why not just have this property use `$ref` to `VectorSearchCompressionConfiguration`? What you have doesn't produce a type-safe and compiled-checked SDK.

Reply:
the profiles object aggregates all the vector features a customer may want to bundle together. So in profiles, they can reference algorithm config named "my-custom-hnsw", add a compression config they named "my-scalar-quantization", etc.

Here in the profiles object, they reference the name they've set, so it's not a discriminated field because it's customer defined.

we use this pattern for custom analyzers/normalizers as well.